libraries/Wire/Wire.cpp

```diff
@@ -404,26 +410,23 @@ void TwoWire::begin(void) {
 /* -------------------------------------------------------------------------- */
 void TwoWire::begin(uint16_t address) {
 /* -------------------------------------------------------------------------- */  
+  end();
   is_master = false;
   slave_address = address;
   /* Address is set inside begin() using slave_address member variable */
-  begin();
+  _begin();
   
 }
 
 /* -------------------------------------------------------------------------- */
 void TwoWire::begin(int address) {
 /* -------------------------------------------------------------------------- */  
-  is_master = false;
```

Comment:
Is this change necessary? What happens in case the caller requests slave -> master -> slave ?

Reply:
This is a simplification (I removed duplicated code). Before there were 3 different `begin(address)` functions and all of them were setting is_master = false, but in the end the all finished up to call the version `begin(uint16_t address)`. So I removed 2 lines of code from the other 2 two (master = false and slave_address = address). You can change role as many time as you want.